### PR TITLE
JSAEM2-11 feat: apply filter to transaction view

### PR DESCRIPTION
**Pull request by MichaelCTH**

### CODE REVIEW

File: src/Stats/utils.js [hunk 1]
```diff
@@ -58,6 +58,14 @@ const filterData = (dataList, range, view) => {
   return result;
 };
 
+const filterType = (dataList, type) => {
+  if (type === 'all') {
```
Comment (ikarasz) [inline src/Stats/utils.js]: use ternary operator:
```js
return type === 'all'
  ? dataList
  : ...
```

File: src/Stats/utils.js [hunk 2]
```diff
@@ -58,6 +58,14 @@ const filterData = (dataList, range, view) => {
   return result;
 };
 
+const filterType = (dataList, type) => {
```
Comment (ikarasz) [inline src/Stats/utils.js]: can we call it `filterTransactionByType` to be honest a stranger would never figure out what's going on in the application if we are full of `filterData`, `filterListView`, `filterType` and similar meaningful function names. But you won't know neither when you open the source code after 3 months.

File: src/Stats/index.jsx [hunk 3]
```diff
@@ -51,12 +53,16 @@ export default function Stats() {
...
+              transFilter={transFilter}
+              setTransFilter={setTransFilter}
+              groupTransactions={filterListView(transactions)}
+            />
           </View>
           {(transactions.length !== 0)
             ? (
               <TransList
-                transactions={filterListView(utils.groupTransactionsByDate(transactions))}
+                transactions={filterListView(utils.filterType(transactions, transFilter))}
```
Comment (ikarasz) [inline src/Stats/index.jsx]: Let's rename the `filterListView` function to `filterTransactions` this method will apply all filters, like type and date. Before rendering you can call this method, which returns all the transactions what we want to display in a flat array. So with this array you can calculate the sum of expenses/incomes which can be passed down to the `FilterBtn` component (So you can remove the calculation from that component) Finally after you calculated the sums you can group the transactions by date and pass it the to `TransList` component.
```js
function filterTransactions(transactions) {
  const transactionsWithSpecificType = filterTransactionsByType(transactions);
  return filterTransactionsByDate(transactionsWithSpecifictype);
}

function calculateSumByType(transactions, type) {
  ...
}

const filteredTransactions = filterTransactions(transactions);
const totalfExpense = calculate...
const totalIncome = calculate...
const processedTransactions = groupTransactionsByDate(filteredTransactions);

....
```

File: src/Stats/index.jsx [hunk 4]
```diff
@@ -51,12 +53,16 @@ export default function Stats() {
       <ScrollView style={styles.deviceBody}>
         <View style={{ marginBottom: 20 }}>
           <View style={{ flexDirection: 'row', justifyContent: 'space-between' }}>
-            <FilterBtn />
+            <FilterBtn
+              transFilter={transFilter}
```
Comment (ikarasz) [inline src/Stats/index.jsx]: lets call this props `currentFilter` or `appliedFilter`

File: src/Stats/index.jsx [hunk 5]
```diff
@@ -51,12 +53,16 @@ export default function Stats() {
...
+              setTransFilter={setTransFilter}
```
Comment (ikarasz) [inline src/Stats/index.jsx]: this property should be `onFilterChnage` and we shouldn't pass setTrandFilter to the component, we should call it in this component instead
```js
onFilterChange={(newFilter) => setTransFilter(newFilter)}
```

File: src/Stats/utils.js (hunk 1 above)
Comment (MichaelCTH) [reply src/Stats/utils.js]: fixed

File: src/Stats/utils.js (hunk 2 above)
Comment (MichaelCTH) [reply src/Stats/utils.js]: fixed

File: src/Stats/index.jsx (hunk 5 above)
Comment (MichaelCTH) [reply src/Stats/index.jsx]: fixed

File: src/Stats/index.jsx (hunk 4 above)
Comment (MichaelCTH) [reply src/Stats/index.jsx]: fixed

File: src/Stats/index.jsx (hunk 3 above)
Comment (MichaelCTH) [reply src/Stats/index.jsx]: fixed

File: src/Stats/index.jsx [hunk 6]
```diff
@@ -17,18 +17,29 @@ export default function Stats() {
...
   const updateHeaderView = (type) => {
     setCurrentView(type);
     setTimePeriod(utils.getDateSet(moment(), type));
     setOverlayVisibility(!isOverlayVisible);
   };
 
-  const filterListView = (transactionRecords) => {
-    const current = timePeriodOptions[1];
-    return utils.filterData(transactionRecords, current, view);
+  const filterTransactions = (transaction) => {
```
[the remark this answers is not in the document]
Comment (MichaelCTH) [reply src/Stats/index.jsx]: parameter name `transactions` conflicts with the variable `transactions` from redux store, on line #16. So will call it `transactionsList` in the new commit

File: src/Stats/index.jsx [hunk 7]
```diff
@@ -17,18 +17,29 @@ export default function Stats() {
...
+    const transactionsWithSpecificType = utils.filterTransactionByType(transaction, transFilter);
```
[the remark this answers is not in the document]
Comment (MichaelCTH) [reply src/Stats/index.jsx]: see above comment

File: src/Stats/index.jsx [hunk 8]
```diff
@@ -17,18 +17,29 @@ export default function Stats() {
...
+    return utils.filterTransactionsByDate(transactionsWithSpecificType, timePeriodOptions[1], view);
   };
 
+  const calculateSumByType = (dataList, type) => dataList
+    .filter((item) => item.type === type)
+    .map((item) => item.amount)
+    .reduce((pre, cur) => parseFloat(pre) + parseFloat(cur), 0);
```
[the remark this answers is not in the document]
Comment (MichaelCTH) [reply src/Stats/index.jsx]: fixed

**Pull request merged**